### DISCUSSION TITLE
the devoicing diacritic And the voicing diacritic. 濁点と半濁点の検索が正常に動作しない

### DISCUSSION THREAD

**Discussion by ShuuyaMonzen** [orchidjs/tom-select#491] [Q&A] [answered]
Please let me know one workaround.
The search is not working correctly when there is a Japanese voiced and semi-voiced point.
Japanese has both full-width and half-width characters, so we think it is difficult to handle all patterns. So it would be good if we could overwrite the search method. Please tell us how to do this.
Thank you from Japan for this project.

**Comment by oyejorge** [comment, marked as answer]
Could you provide some more information about the issue with full-width and half-width characters? Something like "searching with a value of x does not find value of y"?

Customizing search was discussed briefly here: https://github.com/orchidjs/tom-select/discussions/482

**Reply by ShuuyaMonzen** [reply]
Search for "ア" for "アプリ" and you will get hits.
Search "アプ" for "アプリ" and you will not get hits.

Thank you for the workaround.
I was able to find a workaround myself. (I have included ts-ignore for another problem, which is an error in typescript)

var tomselect = new TomSelect(el as TomInput, {});
// @ts-ignore
tomselect.settings.score = function (search) {
        var score = tomselect.getScoreFunction(search);
        return function (item) {
            var searchStr = search;
            return score(item) + item.text.indexOf(searchStr);
        }
}

**Reply by oyejorge** [reply]
> Search for "ア" for "アプリ" and you will get hits.
> Search "アプ" for "アプリ" and you will not get hits.

Try v2.2.2, it should work better

**Reply by ShuuyaMonzen** [reply]
sorry for the late reply
Thank you!!
Your upgrade is working as I expected!

**Comment by ShuuyaMonzen** [comment]
I can clear this problem. thank you!!!
var tomselect = new TomSelect(el as TomInput, {});
        // @ts-ignore
        tomselect.settings.score = function (search) {
            var score = tomselect.getScoreFunction(search);
            return function (item) {
                var searchStr = search;
                return score(item) + (item.text.indexOf(searchStr) + 1);
            };
        }
